### PR TITLE
MAINT: cibuildwheel

### PR DESCRIPTION
**Pull request by larsoner**
This is blind copy-paste so let's see how it goes. Steps:

- [x] Set up azure-pipelines.yml in `master` (just does hello world in Ubuntu, passed in 10 sec)
- [x] Made this PR to modify `azure-pipelines` to do what is suggested by https://github.com/joerick/cibuildwheel#minimal-setup
- [x] Get to produce a wheel
- [x] Bundle fonts with wheels (probably)
- [x] ~~Bundle freetype with wheels on Windows (probably; #1327)~~ #1664
- [x] Get tests to pass (GUIs are a nightmare, skip them ~~maybe by installing `pyqt5` alongside before tests, somehow~~)
   - [x] Windows
   - [x] OSX
   - [x] Linux

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@larsoner Took the chance to correct the closing paranthesis.

**Comment by djhoese** [comment]
Are these configured to automatically upload to PyPI too or do we have to manually do it?

**Comment by larsoner** [comment]
No auto config, with the frequency of our releases I figure manual is okay (and faster to set up once this actually works)

File: setup.py [hunk 1]
```diff
@@ -317,6 +268,7 @@ def run(self):
         'vispy': [op.join('io', '_data', '*'),
                   op.join('html', 'static', 'js', '*'),
                   op.join('app', 'tests', 'qt-designer.ui'),
+                  op.join('util', 'fonts', 'data', '*.ttf'),
```
Comment (djhoese) [inline setup.py]: Do these need to be added to `MANIFEST.in` to be included in the source distribution?

**Comment by larsoner** [comment]
Okay modified `manifest.in`. Last time I ran this, Azure actually passed, so hopefully it continues that way!

**Comment by larsoner** [comment]
Just Travis size check failing

@djhoese can you see the artifacts here?

https://dev.azure.com/vispy/vispy/_build/results?buildId=32&view=results

If not, let me know the email you're using for Azure/MS

**Comment by djhoese** [comment]
I do see the artifacts. Let's merge this and see if we can release this.

File: MANIFEST.in [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
 recursive-include doc/_templates *
+
 recursive-include vispy/static *.*
+recursive-include vispy *.py
+recursive-include vispy *.pyx
+recursive-include vispy *.glsl
+recursive-include vispy *.frag
+recursive-include vispy *.vert
+
+recursive-include js *.*
```
Comment (djhoese) [inline MANIFEST.in]: No! No! No! No! This ends up adding all `node_modules`.

File: MANIFEST.in [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+exclude codegen
+recursive-exclude codegen *
+exclude Makefile
+exclude appveyor.yml
+exclude azure-pipelines.yml
+exclude rtd_requirements.txt
+exclude stdeb.cfg
+exclude .spyderproject
+include .gitmodules
+recursive-exclude vispy *.c
```
Comment (djhoese) [inline MANIFEST.in]: Is there a reason C files *can't* be included?

File: MANIFEST.in [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+
+exclude codegen
+recursive-exclude codegen *
+exclude Makefile
+exclude appveyor.yml
+exclude azure-pipelines.yml
+exclude rtd_requirements.txt
+exclude stdeb.cfg
+exclude .spyderproject
+include .gitmodules
```
Comment (djhoese) [inline MANIFEST.in]: Are you sure anything related to git should be included?

File: MANIFEST.in (hunk 2 above)
Comment (larsoner) [reply MANIFEST.in]: Wasn't sure the right thing to do here

File: MANIFEST.in (hunk 3 above)
Comment (larsoner) [reply MANIFEST.in]: Maybe NumPy headers matter (?)

File: MANIFEST.in (hunk 4 above)
Comment (larsoner) [reply MANIFEST.in]: I think this was the submodule defs. I wasn't really confident in most of these...

File: MANIFEST.in (hunk 2 above)
Comment (djhoese) [reply MANIFEST.in]: Yeah, I'm trying to figure out what might be best here. Was just surprised when building the sdist when a ton of JS libraries were added. 😉

Comment (larsoner) [reply MANIFEST.in]: FWIW the wheels are only 1.5 MB IIRC so it's not so bad

File: MANIFEST.in (hunk 3 above)
Comment (djhoese) [reply MANIFEST.in]: Actually I think C files are required so you can build the .so's from them when installing from sdist.

Comment (larsoner) [reply MANIFEST.in]: Only if they don't have Cython, which we currently list as `build_requires` (and I scrapped all the code to hack/work around this). Feel free to put it back if you want, but to me it felt worth it to simplify things now that wheels will be distributed

Comment (djhoese) [reply MANIFEST.in]: Fair enough.

**Comment by kmuehlbauer** [comment]
Great work @djhoese and @larsoner!

**Pull request merged**